test/jdk/javax/swing/JInternalFrame/InternalFrameTitleButtonTest.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8140527
+ * @key headful
+ * @requires (os.family == "windows")
+ * @summary InternalFrame has incorrect title button width
+ * @run main InternalFrameTitleButtonTest
+ */
+
+import java.awt.Component;
+import java.awt.Insets;
+import java.awt.Robot;
+import javax.swing.Icon;
+import javax.swing.JButton;
+import javax.swing.JDesktopPane;
+import javax.swing.JInternalFrame;
+import javax.swing.JFrame;
+import javax.swing.plaf.basic.BasicInternalFrameTitlePane;
+import javax.swing.plaf.basic.BasicInternalFrameUI;
+import javax.swing.SwingUtilities;
+import javax.swing.UIManager;
+
+public class InternalFrameTitleButtonTest {
+
+    private static JFrame frame;
+    private static JInternalFrame iframe;
+
+    public static void main(String[] args) throws Exception {
+        String osName = System.getProperty("os.name");
+        if(!osName.toLowerCase().contains("win")) {
```

Comment:
@prsadhuk The string compare can be replaced with `OSInfo.getOSType() == OSInfo.OSType.WINDOWS`  and you want to retain the OSCheck along with `@requires (os.family == "windows")` in case the it is run as standalone test.

Reply:
I think [it's redundant](https://github.com/openjdk/jdk/pull/14555#discussion_r1242206200) in this case because setting Windows Classic L&F will throw an exception on other platforms.

It was [`OSInfo`-check initially](https://github.com/openjdk/jdk/pull/14555#discussion_r1240155464).

I'm fine with either way. However, I prefer *no check* in this particular case.

Reply:
Actually I will prefer gentle nudge stating it's not applicable other than windows as compared to harsh failing via throwing exception, if you dont mind..